components/movementsensor/merged/merged.go

```diff
@@ -0,0 +1,373 @@
+
+	if posChanged {
+		m.pos, err = firstGoodSensorWithProperties(deps,
+			newConf.Position,
+			&movementsensor.Properties{PositionSupported: true})
+		if err != nil {
+			return err
+		}
+		m.posReporting = true
+	}
+
+	if compassChanged {
+		m.compass, err = firstGoodSensorWithProperties(deps,
+			newConf.CompassHeading,
+			&movementsensor.Properties{CompassHeadingSupported: true})
+		if err != nil {
+			return err
+		}
+		m.compassReporting = true
+	}
+
+	if lvelChanged {
+		m.linVel, err = firstGoodSensorWithProperties(deps,
+			newConf.LinearVelocity,
+			&movementsensor.Properties{LinearVelocitySupported: true})
+		if err != nil {
+			return err
+		}
+		m.linVelReporting = true
+	}
+
+	if avelChanged {
+		m.angVel, err = firstGoodSensorWithProperties(deps,
+			newConf.AngularVelocity,
+			&movementsensor.Properties{AngularVelocitySupported: true})
+		if err != nil {
+			return err
+		}
+		m.angVelReporting = true
+	}
+
+	if laccChanged {
+		m.linAcc, err = firstGoodSensorWithProperties(deps,
+			newConf.LinearAcceleration,
+			&movementsensor.Properties{LinearAccelerationSupported: true})
+		if err != nil {
+			return err
+		}
+		m.linAccReporting = true
+	}
+
+	return nil
+}
+
+func (m *merged) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+	if m.pos == nil {
+		return geo.NewPoint(math.NaN(), math.NaN()), math.NaN(),
+			movementsensor.ErrMethodUnimplementedPosition
+	}
+
+	return m.Position(ctx, extra)
+}
+
+func (m *merged) Orientation(ctx context.Context, extra map[string]interface{}) (spatialmath.Orientation, error) {
+	if m.ori == nil {
+		nanOri := spatialmath.NewOrientationVector()
+		nanOri.OX = math.NaN()
+		nanOri.OY = math.NaN()
+		nanOri.OZ = math.NaN()
+		nanOri.Theta = math.NaN()
+		return nanOri,
+			movementsensor.ErrMethodUnimplementedOrientation
+	}
+	return m.Orientation(ctx, extra)
+}
+
+func (m *merged) CompassHeading(ctx context.Context, extra map[string]interface{}) (float64, error) {
+	if m.compass == nil {
+		return math.NaN(),
+			movementsensor.ErrMethodUnimplementedCompassHeading
+	}
+	return m.compass.CompassHeading(ctx, extra)
+}
+
+func (m *merged) LinearVelocity(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	if m.linVel == nil {
+		return r3.Vector{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedLinearVelocity
+	}
+	return m.linVel.LinearVelocity(ctx, extra)
+}
+
+func (m *merged) AngularVelocity(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+	if m.angVel == nil {
+		return spatialmath.AngularVelocity{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedAngularVelocity
+	}
+	return m.angVel.AngularVelocity(ctx, extra)
+}
+
+func (m *merged) LinearAcceleration(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	if m.linAcc == nil {
+		return r3.Vector{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
```

Comment:
These are very verbose ways of returning errors. Could we `return nil, movementsensor.Err...` instead? "No" is an okay answer; I'm just asking.

Reply:
I like how explicit `math.NaN` is, I also like how this won't cause a panic or nil pointer when trying to access a field in `*geo.Point* and `spatialmath.Orientation/AngularVelocity` since those are interfaces with methods assigned to that struct if you use the go SDK to do more math upstream of this driver. But if you feel like this is overkill I can change them all to `nil`.